### PR TITLE
feat: rework asset-manager for better timing support

### PR DESCRIPTION
**Pull request by mattstrom**
[RKT-8830]

### CODE REVIEW

File: src/runner.js [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    };
+
+    /**
+     * @returns void
+     * @private
+     */
+    Runner.prototype.confirm = function () {
+        const client = this.container.client;
+        // Q: Could have been a bug? Only confirming on first occurrence, not subsequent context entries
+        if (!this.confirmed) {
```
Comment (robertsevernsentient) [inline src/runner.js:317]: I think this should be removed -- if it's left in then on an SPA we won't try to confirm a new context if the page isn't reload

File: src/tests/runner.test.js [hunk 2]
```diff
@@ -0,0 +1,304 @@
+import * as assert from 'assert';
```
Comment (robertsevernsentient) [inline src/tests/runner.test.js:1]: Plenty more cases I want us to unit test here -- but enough to get deployed for manual testing

**Review by robertsevernsentient — COMMENTED** [review]
hotfix-v1.8

**Review by robertsevernsentient — CHANGES_REQUESTED** [review]
We need to understand why the index.js tests are failing before we can merge and deploy

**Comment by mattstrom** [comment]
> We need to understand why the index.js tests are failing before we can merge and deploy

@robertsevernsentient: I've corrected nearly all of the failing tests, except for this one which I haven't been able to discern the intent of yet. The test fails because `invokedJavascript.length` equals 6 instead of 7. There's a nearly identical block just above it.

Any thoughts? Maybe there's supposed to be a redundant invocation:
https://github.com/evolv-ai/asset-manager/blob/1c3ff2b7e6e157137dd8a6f034c37f0e5b4447cb/src/tests/index.test.js#L539-L546

**Comment by robertsevernsentient** [comment]
> > We need to understand why the index.js tests are failing before we can merge and deploy
> 
> @robertsevernsentient: I've corrected nearly all of the failing tests, except for this one which I haven't been able to discern the intent of yet. The test fails because `invokedJavascript.length` equals 6 instead of 7. There's a nearly identical block just above it.
> 
> Any thoughts? Maybe there's supposed to be a redundant invocation:
> https://github.com/evolv-ai/asset-manager/blob/1c3ff2b7e6e157137dd8a6f034c37f0e5b4447cb/src/tests/index.test.js#L539-L546

Looking into it now @mattstrom -- basic issue is that 'evolv_web_page1' wasn't marked resolved -- so didn't get marked as runnable again when the key was removed

**Comment by robertsevernsentient** [comment]
> > > We need to understand why the index.js tests are failing before we can merge and deploy
> > 
> > 
> > @robertsevernsentient: I've corrected nearly all of the failing tests, except for this one which I haven't been able to discern the intent of yet. The test fails because `invokedJavascript.length` equals 6 instead of 7. There's a nearly identical block just above it.
> > Any thoughts? Maybe there's supposed to be a redundant invocation:
> > https://github.com/evolv-ai/asset-manager/blob/1c3ff2b7e6e157137dd8a6f034c37f0e5b4447cb/src/tests/index.test.js#L539-L546
> 
> Looking into it now @mattstrom -- basic issue is that 'evolv_web_page1' wasn't marked resolved -- so didn't get marked as runnable again when the key was removed

I had to change unschedule to successfully unschedule resolved functions too

**Review by robertsevernsentient — APPROVED** [review]

**Comment by robertsevernsentient** [comment]
hotfix-v1.8

**Pull request merged**